source/docs/software/wpilib-tools/wpical/index.rst

```diff
@@ -0,0 +1,177 @@
+
+.. image:: images/ChessboardDetection.png
+    :alt: CheckerboardDetection
+
+.. warning:: If any frames from the camera calibration look suspect, take a new video of the calibration board and try again.
+
+The camera calibration will automatically load the generated camera intrinsics JSON into WPIcal to continue to field calibration, but will also output the JSON file so it can be used for future calibrations.
+
+Upload Calibration JSON
+^^^^^^^^^^^^^^^^^^^^^^^
+
+As opposed to calibrating cameras in the tool, WPIcal also allows you to upload a camera intrinsics JSON. There are a three things each calibration JSON needs:
+
+* ``avg_reprojection_error``
+* ``camera_matrix``
+* ``distortion_coefficients``
+
+Example:
+
+.. code-block:: json
+    
+    {
+        "avg_reprojection_error": 0.3989609373420966,
+        "camera_matrix": [
+            1767.912766105077,          // fx
+            0.0,
+            989.4346078930205,          // cx
+            0.0,
+            1763.5161174120185,         // fy
+            540.7047848580446,          // cy
+            0.0,
+            0.0,
+            1.0
+        ],
+        "distortion_coefficients": [
+            0.3878712310541597,         // k1
+            -2.565991238351721,         // k2
+            -0.0011767048240595168,     // p1
+            0.0017341455009923185,      // p2
+            5.70562560518866,           // k3
+            0.03694246896004978,        // k4
+            0.11811204683668342,        // k5
+            -0.2274481545065017,        // k6
+            0.0,                        // s1
+            0.0,                        // s2
+            0.0,                        // s3
+            0.0,                        // s4
+            0.0,                        // tx
+            0.0                         // ty
+        ]
+    }
+
+
+One common method for external camera calibration is `CalibDB.net <https://calibdb.net/>`_. WPIcal allows you to directly upload the generated CalibDB JSON without any modification. Make sure to download the OpenCV format JSON:
+
+
+.. image:: images/CalibdbDownload.png
+    :alt: CalibdbDownload
+
+Then:
+
+.. image:: images/CalibdbOpenCVFormat.png
+    :alt: CalibdbOpenCVFormat
+
+.. important:: When uploading a CalibDB JSON, ensure that you download the calibration with the proper resolution selected.
+
+.. warning:: CalibDB calibrations have been known to fail when bad snapshots are taken. If your calibration data looks suspect, repeat calibration process again.
+
+Field Calibration
+-----------------
+
+After calibrating the camera, you can use the camera model to find the relative positions of the Apriltags. The calibration process will generate a .json file and a .fmap for use on coprocessors and in robot code.
+
+Upload Ideal Field Map
+^^^^^^^^^^^^^^^^^^^^^^
+
+WPIcal uses an "ideal" field map JSON as an initial guess point for the optimization. It is recommended to upload the json file included with WPILib.
```

Comment:
Does the tool make it easy to upload an existing WPILIb field JSON file, or does there need to be more instructions on where to get one.